e2e/test-reload.py

```diff
@@ -32,12 +32,8 @@ def main():
         'e2e/repeat-action/build/libs/repeat-action.jar'
     ])
 
-    # We're expecting the startup to take between 40ms and 6s. We've seen
-    # 6s in the wild, e.g. on OSX resolving host-names. The app will produce
-    # events for 10s after startup, so we'd expect between 2.5 and 8.5s of
-    # events from the app. We only need 0.5s, so this is a pretty good safety
-    # margin.
-    sleep(7.5)
+    # app outputs events for 4 seconds, and takes under a second to start
+    sleep(1)
```

Comment:
would it be feasible to `sleep` until `something` happens instead of until 1 second passes?
I mean, why are we `sleep`ing here at all?

Reply:
The app is running on a fixed schedule, so outputs eight events, one every 500ms. We want those events to have happened, so we're pretty convinced it will have happened after a whole second (ages!).

I think we discussed changing the test app before, to generate events/http posts/new files, etc., and decided it wasn't worth complicating the test app. Remember that the agent has to deal with all code in the test app (even if it's not production code), so things are easier to debug when the test app is tiny; the purpose of these tests.